packages/testing/src/factories/mock-db-client.ts

```
@@ -0,0 +1,191 @@
+  ];
+
+  // Create all chain methods
+  chainMethods.forEach((method) => {
+    (mockBuilder as any)[method] = vi.fn().mockReturnValue(mockBuilder);
+  });
+
+  // Terminal method that returns a promise
+  mockBuilder.returning = vi.fn().mockResolvedValue(finalValue);
+
+  // Make the builder awaitable (like Drizzle queries)
+  mockBuilder.then = vi.fn((resolve) => Promise.resolve(resolve(finalValue)));
```

Biome (2.1.2)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)